### DISCUSSION TITLE
Rapid repeat connections from Zigbee2MQTT to Sonoff ZBBridge results in Tasmota reboot and configuration loss

### DISCUSSION THREAD

**Discussion by Heligear** [Koenkk/zigbee2mqtt#17219] [General]
Firstly, thanks for reading. I am trying to understand which steps I take take to help diagnose the issue I am experiencing with Zigbee2MQTT. 
I am using Zigbee2MQTT as it works better (or just works) with some devices that don't work with Zigbee Home Automation.

**The Sonoff ZHABridge Tasmota is configured using the following steps and is found and connects devices OK**

**Device configuration:**
[Connect to a remote Sonoff ZBBridge | Zigbee2MQTT](https://www.zigbee2mqtt.io/advanced/remote-adapter/connect_to_a_remote_sonoff_zbbridge.html)
Which follows this post through to step 6 [How to use the Sonoff Zigbee Bridge with Home Assistant - Tasmota | digiblurDIY](https://digiblur.com/2020/07/25/how-to-use-the-sonoff-zigbee-bridge-with-home-assistant-tasmota/)

**Sonoff ZNZB-02 Versions - Tasmota Zigbee2MQTT Gateway
Sonoff ZHABridge Tasmota**
Program Version	12.4.0.4(zbbridge)
Build Date & Time	2023-03-30T06:58:25
Core/SDK Version	2_7_4_9/2.2.2-dev(38a443e)
Uptime	0T00:04:26
Flash Write Count	121 at 0x1FB000
Boot Count	59
Restart Reason	Software/System restart

**Running HomeAssistant OS on Hyper-V VM**
3GB memory configured (Usage typically between 700MB and 2.4GB)
Home Assistant OS/Core/Supervisor are all up to date

**Zigbee2MQTT**
Version 1.30.3-1
Start on boot - Enabled
Watchdog - Enabled
Auto Update - Disabled
Show on sidebar - Enabled

serial:
  port: tcp://x.x.x.x:8888
  adapter: ezsp

MQTT
base topic: zigbee2mqtt
MQTT Server: mqtt://core-mosquitto:1883

**Issue:**
After a period of running successfully (10-120 minutes), Zigbee2MQTT will start rapidly connecting to the Tasmota Gateway. 500-750 times over a 3 minute period resulting the Tasmota Gateway rebooting.

![Zigbee2MQTT issue](https://user-images.githubusercontent.com/129662504/229350066-2a8f09a0-0206-424e-969d-b0334c028c0c.png)

This repeats between 500-750 times in 3 minutes which results in the Tasmota Gateway rebooting.
When the Tasmota gateway reboots port 8888 does not come back online.

**The only way to get port 8888 back online is to re-run the command from the configuration.**

**#6 Issue the following command on the console to map the Zigbee chip to TCP Server Port 8888:**
backlog rule1 on system#boot do TCPStart 8888 endon ; rule1 1 ; template {"NAME":"Sonoff ZHABridge","GPIO":[56,208,0,209,59,58,0,0,0,0,0,0,17],"FLAG":0,"BASE":18} ; module 0

**Once the console reboots you should see the server has started on Port 8888:**
17:35:58 TCP: Starting TCP server on port 8888
17:35:58 RSL: stat/tasmota_7FC5B0/RESULT = {"TCPStart":"Done"}

With Zigbee2MQTT watchdog enable it will normally reconnect but within 10-120 minutes the cycle starts again where by the Tasmota gate get the high volume of connections from Home Assistant Zigbee2MQTT and the Tasmota Gateway reboots and I am back to square 1.

**So I think I have 2 issues, the second  is most like die to the first.**
1. Zigbee2MQTT has "an issue" that causes it to perform multiple reconnects (this happens if watchdog is enabled or not, which suggests to me that Zigbee2MQTT is not crashing an restarting rapidly)
	
2. The Sonoff ZB Bridge "losses" its TCP port 8888 configuration after presumably falling over after being "swapped" by connections from Zigbee2MQTT. I believe this is just a consequence of issue 1 and am not that concerned about it, but I thought I would report the observation.
	
What I need help with is what steps to take to help diagnose what is causing Zigbee2MQTT to create rapid connections to the Tasmota gateway. Before this happens the 2 devices connected via Zigbee2MQTT are reporting ok with issues.
I also have a separate Sonoff ZB Bridge running Tasmota for Zigbee Home Automation with a number of devices connected without issue. I am using Zigbee2MQTT as well as the Power Clamp and the Tuya Temp sensors don't work properly with Zigbee Home Automation.

**Comment by thestraycat** [comment]
@Heligear - My experience today was exactly the same, i wish i'd seen this post sooner rathern than troublehsooting it! 

I too noticed that restarting the TCP server on port 8888 on the bridge got everything back up. Is there no way to make starting TCP server on port 8888 persistant i wonder on the bridge? Pretty annoying if that needs to be run every time of an outage.

Just to add a little more info to this issue, i am running a very similar setup (different vm spec and platform is all really, everything else identical to Heligear.) Other than i only have 2 devices connected to my Zigbee2MQTT setup, (2 x Zigbee Candeo Dimmers) both act as routers so my zigbee network is tiny. the LQI is also pretty reasonable. See below.

Am i right in thinking the Sonoff ZBBridge has even been since removed from the experiemental adaptor section on the Zigbee2MQTT recommended adaptor page? 

![image](https://user-images.githubusercontent.com/9296295/231827351-8b6b28f1-7238-40e5-b141-0283c1d5da1d.png)



**Sonoff ZNZB-02 Versions - Tasmota Zigbee2MQTT Gateway**
**Sonoff ZHABridge Tasmota**
Program Version 12.4.0.4(zbbridge)
Build Date & Time 2023-03-30T06:58:25
Core/SDK Version 2_7_4_9/2.2.2-dev(38a443e)
Flash Write Count 73 at 0x1FB000
Boot Count 42
Restart Reason Software/System restart


**Running HomeAssistant OS on KVM Unraid**
2GB memory configured 
2vcpu
32gb disk
Home Assistant OS/Core/Supervisor are all up to date

**Zigbee2MQTT**
Version 1.30.3-1
Start on boot - Enabled
Watchdog - Enabled
Auto Update - Disabled
Show on sidebar - Enabled

**serial:**
port: tcp://x.x.x.x:8888
adapter: ezsp

**MQTT**
base topic: zigbee2mqtt
MQTT Server: mqtt://core-mosquitto:1883

**Comment by thestraycat** [comment]
@Heligear - I take it you've seen this post? I believe its likely just as valid for Z2MQTT... 

https://github.com/home-assistant/home-assistant.io/issues/17170
